doc/dev/architecture.rst

```diff
@@ -5,71 +5,392 @@ Architecture
 
    <br />
 
-Overview
-^^^^^^^^
+The Catalyst stack leverages state-of-the-art technologies to accelerate quantum workflows without
+losing the ability to quickly prototype in Python. Some of the projects Catalyst builds upon include
+the `MLIR <https://mlir.llvm.org/docs/>`_ and `LLVM <https://llvm.org/>`_ compiler frameworks, the
```

Comment:
```suggestion
losing the ability to quickly prototype in Python. To do so, Catalyst builds upon include
the `MLIR <https://mlir.llvm.org/docs/>`_ and `LLVM <https://llvm.org/>`_ compiler frameworks, the
```

Reply:
I don't think the new sentence makes grammatical sense 😅

Reply:
Oops! Yes I meant to also remove include!